cranelift/filetests/filetests/isa/aarch64/simd-logical-compile.clif

```diff
@@ -22,7 +22,7 @@ block0(v0: b32x4):
 ;   umaxp v3.4s, v0.4s, v0.4s
 ;   mov x5, v3.d[0]
 ;   subs xzr, x5, #0
-;   csetm x0, ne
+;   cset x0, ne
```

Comment:
This change switches from an output of all-ones to just 1 for true -- I guess that's OK, since the return type is `b1`? We have a separate TODO to check over our bool representation consistency anyway so I guess this is fine for now.

Reply:
Yes - `materialize_bool_result` returns all 1s/0s for the length of the output type - so this happens because the return type is `b1`.

Reply:
@cfallin This used to be a harmless mistake in the previous implementation of the `vany_true` operation, which checked the input type to determine the width of the output boolean.

Reply:
Makes sense, thanks!